addons/mass_mailing/static/src/js/mailing_filter_widget.js

```diff
@@ -80,12 +81,19 @@ const FieldMailingFilter = FieldMany2One.extend({
      * @private
      */
     _updateFilterIcons: function () {
-        const hasDomain = this.recordData[this.domainFieldName] && this.recordData[this.domainFieldName] !== '[]';
-        this.el.querySelector('.o_mass_mailing_filter_container').classList.toggle('invisible', !hasDomain);
+        const filterCount = this.recordData.mailing_filter_count;
+        // Show custom message when no filter is available for selected model
+        this.el.querySelector(".o_mass_mailing_no_filter").classList.toggle('d-none', filterCount);
+        this.el.querySelector(".o_input_dropdown").classList.toggle('d-none', !filterCount);
```

Comment:
can you use this css selector instead ? `.o_mass_mailing_filter_container .o_input_dropdown`

(just to be sure to select the right element, e.g. if we add other widget in the futur with this class)

Reply:
Actually we want to hide the input coming from original m2o widget, while `.o_mass_mailing_filter_container` is a sibling of `.o_input_dropdown`, not a parent, however, I have used a more robust selector :)